torch/_decomp/decompositions.py

```diff
@@ -3042,6 +3080,34 @@ def _rnn_helper(
     return input, final_hiddens
 
 
+# Equivalent to C++ except that it calls slice_strict.  We only override it in
+# Python dispatcher to reduce blast radius of change
```

Comment:
this strategy seems fine, but - mostly a curiosity question on how you chose an implementation - if you goal is to simplify boundary condition handling for slice while minimizing blast radius, what do you think of having the existing `aten.slice` unconditionally change its behavior, only when `start/length` are unbacked symints, to raise runtime asserts?

After typing that, I guess that would give our slice/narrow ops some pretty annoying-to-define invariants. So a dedicated op that spells out the error checks feels pretty reasonable.

Reply:
Yes, exactly. I did think about doing it this way, but it just sets a bad precedent (different runtime behavior if with unbacked or not) that guard size oblivious doesn't really have.